ansible/roles/lustre/tasks/configure.yml

```diff
@@ -9,9 +9,11 @@
 
 - name: Set facts for Lustre interface
   set_fact:
-    _lustre_interface: "{{ _lustre_ip_r_mgs_info[4] }}"
-    _lustre_ip: "{{ _lustre_ip_r_mgs_info[6] }}"
+    _lustre_interface: "{{ _lustre_ip_r_mgs_info[interface_index | int] }}"
+    _lustre_ip: "{{ _lustre_ip_r_mgs_info[ip_index | int] }}"
   vars:
+    interface_index: "{{ 4 if 'via' in _lustre_ip_r_mgs.stdout_lines.0 else 2 }}"
```

Comment:
Rather than relying on parsing a pretty nasty string, could we generate `_lustre_ip_r_mgs` as JSON from iproute and parse it that way? Tested quickly on RL8 and RL9 iproute packages...

```
ip --json ro get 192.168.178.1
[{"dst":"192.168.178.1","dev":"wlp165s0","prefsrc":"192.168.178.90","flags":[],"uid":0,"cache":[]}]
```

I know its a slightly bigger change but it feels a bit less brittle!

Reply:
Nice suggestion :)

Reply:
Lovely!